### PR TITLE
Feature/search api

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

File: web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * @file
+ * Contains Drupal\itk_footer\Form\ITKFooterContentForm.
```
Comment (tuj) [inline web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php]: ITKFooterContent

File: web/modules/custom/genlyd_search/src/Controller/PageController.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\search_api\Entity\Index;
+use Drupal\taxonomy\Entity\Vocabulary;
+use Drupal\views\Plugin\views\display\Page;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class ApiController.
```
Comment (tuj) [inline web/modules/custom/genlyd_search/src/Controller/PageController.php]: ApiController?

File: web/modules/custom/genlyd_search/src/Controller/ApiController.php [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    if (!empty($hits)) {
+      $json['no_of_results'] = $result->getResultCount();
+      $json['hits'] = $hits;
+    }
+
+    $response =  new JsonResponse($json);
+    return $response;
+  }
+
+  public function searchGeoJSON(Request $request) {
```
Comment (tuj) [inline web/modules/custom/genlyd_search/src/Controller/ApiController.php]: Comment

File: web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+namespace Drupal\genlyd_search\Form;
+
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\taxonomy\Entity\Vocabulary;
+
+/**
+ * Class ITKFooterContentForm
```
Comment (tuj) [inline web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php]: ITKFooterContentForm?

File: web/modules/custom/genlyd_search/src/Plugin/search_api/processor/GeoCoderField.php [hunk 5]
```diff
@@ -0,0 +1,126 @@
+<?php
+
```
Comment (tuj) [inline web/modules/custom/genlyd_search/src/Plugin/search_api/processor/GeoCoderField.php]: @file

File: web/modules/custom/genlyd_search/src/Plugin/search_api/processor/Property/GeoCoderFieldProperty.php [hunk 6]
```diff
@@ -0,0 +1,207 @@
+<?php
+
```
Comment (tuj) [inline web/modules/custom/genlyd_search/src/Plugin/search_api/processor/Property/GeoCoderFieldProperty.php:2]: @file

File: web/modules/custom/genlyd_search/templates/genlyd-search-page.html.twig [hunk 7]
```diff
@@ -0,0 +1,171 @@
+<div data-ng-app="genlyd" data-ng-controller="SearchController">
```
Comment (tuj) [inline web/modules/custom/genlyd_search/templates/genlyd-search-page.html.twig:1]: Should resemble the others twig templates from genlyd

**Review by tuj — CHANGES_REQUESTED** [review]

File: web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php (hunk 1 above)
Comment (cableman) [reply web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php]: Fixed

File: web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php (hunk 4 above)
Comment (cableman) [reply web/modules/custom/genlyd_search/src/Form/GenlydSearchSettingsForm.php]: Fixed

File: web/modules/custom/genlyd_search/src/Controller/PageController.php (hunk 2 above)
Comment (cableman) [reply web/modules/custom/genlyd_search/src/Controller/PageController.php]: Fixed

File: web/modules/custom/genlyd_search/src/Controller/ApiController.php (hunk 3 above)
Comment (cableman) [reply web/modules/custom/genlyd_search/src/Controller/ApiController.php]: Fixed

File: web/modules/custom/genlyd_search/src/Plugin/search_api/processor/GeoCoderField.php (hunk 5 above)
Comment (cableman) [reply web/modules/custom/genlyd_search/src/Plugin/search_api/processor/GeoCoderField.php]: Fixed

File: web/modules/custom/genlyd_search/src/Plugin/search_api/processor/Property/GeoCoderFieldProperty.php (hunk 6 above)
Comment (cableman) [reply web/modules/custom/genlyd_search/src/Plugin/search_api/processor/Property/GeoCoderFieldProperty.php:2]: Fixed

File: web/modules/custom/genlyd_search/templates/genlyd-search-page.html.twig (hunk 7 above)
Comment (cableman) [reply web/modules/custom/genlyd_search/templates/genlyd-search-page.html.twig:1]: Yes, think we have to do this together.

**Pull request merged**